Tests porting: foundation text and text-field

eymar opened the pull request.

eymar committed:
Tests poring: foundation text and text-field

eymar commented on compose/ui/ui-test-junit4/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skikoMain.kt:
```diff
@@ -261,6 +266,16 @@ class SkikoComposeUiTest(
         return surface.makeImageSnapshot()
     }
 
+    fun SemanticsNodeInteraction.captureToImage(): ImageBitmap {
```
API change: new method

igordmn replied:
nit: Can we use `waitForIdle()` + `surface.makeImageSnapshot(iRect)` for better performance?

eymar committed:
temporary Ignore TextFieldUndoTest

igordmn commented on compose/ui/ui-test-junit4/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skikoMain.kt:
```diff
@@ -261,6 +266,16 @@ class SkikoComposeUiTest(
     }
 
+    fun SemanticsNodeInteraction.captureToImage(): ImageBitmap {
+        val rootImage = this@SkikoComposeUiTest.captureToImage()
```
`captureToImage` and `SemanticsNodeInteraction.captureToImage()` have different types (Image, ImageBitmap). We should keep only one. We probably should return `ImageBitmap`, so our API will depend on Compose, not Skia.

It is a breaking change, but our test API is still experimental

eymar replied:
I agree. Done.
Also updated some tests which used `fun captureToImage(): Image`

eymar committed:
Make fun captureToImage return ImageBitmap instead of Image

Image is a type from skiko, and ImageBitmap is a type from Compose (better to expose it)

eymar commented on compose/ui/ui-test-junit4/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skikoMain.kt:
```diff
@@ -256,9 +261,17 @@ class SkikoComposeUiTest(
     }
 
-    fun captureToImage(): Image {
+    fun captureToImage(): ImageBitmap {
```
Breaking API chnage: Return type was Image, now - ImageBitmap
